### PR TITLE
bpo-32604: Recommit "bpo-32604: PEP 554 for use in test suite (GH-19985)"

### PR DESCRIPTION
**Pull request by nanjekyejoannah**
Recommit of what was reverted in [9d17cbf](https://github.com/python/cpython/commit/9d17cbf33df7cfb67ca0f37f6463ba5c18676641)

The refleak in the select module was fixed in a [separate PR](https://github.com/python/cpython/pull/20600#issuecomment-638175474) 

With this fix, the tests pass:
```
gitpod /workspace/cpython $ ./python -m test -R 3:3 test_interpreters
0:00:00 load avg: 4.74 Run tests sequentially
0:00:00 load avg: 4.74 [1/1] test_interpreters
beginning 6 repetitions
123456
......
test_interpreters passed in 1 min 9 sec

== Tests result: SUCCESS ==

1 test OK.

Total duration: 1 min 9 sec
Tests result: SUCCESS
```

https://bugs.python.org/issue32604

### CODE REVIEW

**Comment by vstinner** [comment]
Sorry, I don't have the bandwidth to review this PR.

**Comment by nanjekyejoannah** [comment]
The contents of this PR were already reviewed by @ericsnowcurrently . The only confirmation here is to confirm that no leaks make the tests fail which I already did. There are no changes to what had been merged before

File: Lib/test/support/interpreters.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            for cid in _interpreters.channel_list_all()]
+
+
+_NOT_SET = object()
+
+
+class RecvChannel:
+    """
+    The RecvChannel object represents
+    a recieving channel.
```
Comment (pablogsal) [inline Lib/test/support/interpreters.py]:
```suggestion
    a receiving channel.
```

**Comment by pablogsal** [comment]
I didn't review this but I am confused....why there is an `rst` file in the `test` directory?

**Comment by nanjekyejoannah** [comment]
@pablogsal, I think the discussion around this was we wanted to have a reference to the added API in ``test.support ``. We could not add the document file in ``Doc/library/*`` before the API moved to ``stdlib``. You can follow the discussion here: https://github.com/python/cpython/pull/19985

**Comment by nanjekyejoannah** [comment]
@pablogsal I decided to remove the docs. I hope this is okay now.

**Review by ericsnowcurrently — APPROVED** [review]
LGTM  As I said when I reviewed the original PR, this is a good start. :)  Thanks for doing this.

**Pull request merged**